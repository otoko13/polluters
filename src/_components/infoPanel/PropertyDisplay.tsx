import React, { FunctionComponent } from 'react';
import './propertyDisplay.scss';

export interface IPropertyDisplayProps {
    icon: JSX.Element;
    title: string;
}

const PropertyDisplay: FunctionComponent<IPropertyDisplayProps> = ({ icon, title, children }) => (
    <div className="PropertyDisplay">
        <div className="icon-container" title={title}>
            {icon}
        </div>
        <div className="content">
            <div className="title">
                {title}
            </div>
            {children}
        </div>
    </div>
);

export default PropertyDisplay;