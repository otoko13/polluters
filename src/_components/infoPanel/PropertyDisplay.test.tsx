import { render } from '@testing-library/react';
import * as React from 'react';
import HomeIcon from '@material-ui/icons/Home';
import PropertyDisplay, { IPropertyDisplayProps } from './PropertyDisplay';

let props: IPropertyDisplayProps;

beforeEach(() => {
  props = {
    icon: <HomeIcon />,
    title: 'Home',
  };
});

describe('snapshots', () => {
  it('should look sensible', () => {
    const { container } = render(<PropertyDisplay {...props} />);
    expect(container.firstChild).toMatchSnapshot('State');
  });
});
