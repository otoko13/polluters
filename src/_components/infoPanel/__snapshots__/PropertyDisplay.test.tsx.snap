// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should look sensible 1`] = `
<div
  class="PropertyDisplay"
>
  <div
    class="icon-container"
    title="Home"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
  <div
    class="content"
  >
    <div
      class="title"
    >
      Home
    </div>
  </div>
</div>
`;
